src/System.Distributed.AI.Agents/Utilities/ChatClientExtensions.cs

```diff
@@ -90,15 +90,21 @@ public static ChatClientBuilder AddKeyedChatClient(this IHostApplicationBuilder
         {
             ClientChatProvider.Ollama => builder.AddKeyedOllamaClient(connectionName, connectionInfo),
             ClientChatProvider.OpenAI => builder.AddKeyedOpenAIClient(connectionName, connectionInfo),
-            ClientChatProvider.AzureOpenAI => builder.AddKeyedAzureOpenAIClient(connectionName).AddKeyedChatClient(connectionName, connectionInfo.SelectedModel),
+            ClientChatProvider.AzureOpenAI => builder.AddKeyedAzureOpenAIClient(connectionName, s =>
+            {
+                s.DisableMetrics = false;
+                s.DisableTracing = false;
+            }).AddKeyedChatClient(connectionName, connectionInfo.SelectedModel),
             ClientChatProvider.AzureAIInference => builder.AddKeyedAzureInferenceClient(connectionName, connectionInfo),
             _ => throw new NotSupportedException($"Unsupported provider: {connectionInfo.Provider}")
         };
 
         // Add OpenTelemetry tracing for the ChatClient activity source
-        chatClientBuilder.UseOpenTelemetry().UseLogging();
+        chatClientBuilder
+            //.UseOpenTelemetry()
```

Comment:
Adding this before `UseDurableFunctionInvocation` means that we don't register an `ActivitySource` in "DI" by the time `UseDurableFunctionInvocation` tries to access it.
Used the first class tracing in `AddKeyedAzureOpenAIClient` for now, but seems like we want some generic solution for all chat client types.

We could call `UseOpenTelemetry` after `UseDurableFunctionInvocation` so there will be an `ActivitySource` available.